(DOCSP-21050): Sync roles updates (#1679)
mongodb/docs-realm: source/sync/learn/overview.txt

Flexible Sync Data Access Rules
```````````````````````````````

> Flexible Sync does not require you to define a partition strategy, but
> instead enables you to specify which data to sync by querying for
> matching objects in a client application. Realm then evaluates

Reviewer: Another long sentence break-up:

```suggestion
Flexible Sync does not require you to define a partition strategy.
Instead, you specify which data to sync through queries for
matching objects in a client application. Realm then evaluates
```
